### ISSUE TITLE
Typerror when calling super() in constructor

### ISSUE THREAD

**Issue by latenssi** [jacobheun/timeout-abort-controller#2]
I was debugging an error in [js-libp2p](https://github.com/libp2p/js-libp2p) which caused it to be unable to connect to peers in a production build in browser build by webpack using create-react-app framework.

I finally found this error message in the logs when a connection fails:
`TypeError: Failed to construct 'AbortController': Please use the 'new' operator, this DOM object constructor cannot be called as a function.`

If I replace the implementation of `TimeoutController` with one that uses JS prototypes rather than JS classes (and therefore no super() calls) I get no errors and the connections happen as intended.

I am not sure why this happens only in a production build and not in a development setup (webpack-dev-server) -- I think it has something to do with Typescript and babel and webpack -- but this does happen consistently and prevents libp2p connections in create-react-app based apps. I am also not sure why this error has not surfaced anywhere else but obviously (as the error message states) it is a real bug in the code.

**Comment by jacobheun** [comment]
@latenssi what is your es build target for webpack? I am wondering if the production build is compiling to es5 resulting in a `_super.call(this)` type of call, which might cause this error. The dev server is likely just using es6 targets.

**Comment by latenssi** [comment]
Yes you are correct, my create-react-app setup seems to be using ES 5 as target.
